main.go

```diff
@@ -90,19 +103,200 @@ func newtableCosts(table string, readMetrics, writeMetrics []dynamoDBMetric) (s
-	s.onDemandWrite += 1.4864 * (writes / float64(1000000))
 
-	// Read capacity unit (RCU)	$0.0001544 per RCU
+	// Read request units
+	s.onDemandRead += pricing.onDemandRead * reads
+	// Write request units
+	s.onDemandWrite = pricing.onDemandWrite * writes
+
+	// Read capacity unit (RCU)
 	for _, m := range readMetrics {
-		s.provisionedRead += m.ProvisionedUnits * 0.0001544
+		s.provisionedRead += m.ProvisionedUnits * pricing.readCapacityUnit
 	}
-	// Write capacity unit (WCU)	$0.000772 per WCU
+	// Write capacity unit (WCU)
 	for _, m := range writeMetrics {
-		s.provisionedWrite += m.ProvisionedUnits * 0.000772
+		s.provisionedWrite += m.ProvisionedUnits * pricing.writeCapacityUnit
+	}
+	return
+}
+
+type dynamoPricing struct {
+	location                            string
+	onDemandRead, onDemandWrite         float64
+	readCapacityUnit, writeCapacityUnit float64
+}
+
+type priceDetail struct {
+	Product     product
+	ServiceCode string
+	Terms       terms
+}
+
+type product struct {
+	ProductFamily string
+	Sku           string
+	Attributes    attributes
+}
+
+type attributes struct {
+	Location string
+}
+
+type terms struct {
+	OnDemand map[string]interface{}
+}
+
+type onDemandTerms struct {
+	PriceDimensions map[string]interface{}
+}
+
+type onDemandDetails struct {
+	Description  string
+	PricePerUnit pricePerUnit
+}
+
+type pricePerUnit struct {
+	Usd string
+}
+
+func parseProduct(product aws.JSONValue) (float64, error) {
+	var result priceDetail
+	mapstructure.Decode(product, &result)
+
+	if len(result.Terms.OnDemand) > 1 {
+		return 0, errors.New("More than one product term found")
+	}
+
+	for _, v := range result.Terms.OnDemand {
+
+		var result2 onDemandTerms
+		mapstructure.Decode(v, &result2)
+
+		for _, v2 := range result2.PriceDimensions {
+
+			var result3 onDemandDetails
+
+			mapstructure.Decode(v2, &result3)
+
+			price, err := strconv.ParseFloat(result3.PricePerUnit.Usd, 64)
+
+			if err != nil {
+				fmt.Printf("Couldn't find any on-demand pricing")
+			}
+
+			// Return first non-zero to ignore any free-tier pricing
+			if price > 0.00 {
+				return price, nil
+			}
+
+		}
+	}
+
+	return 0, errors.New("No product data found")
+}
+
+func getDynamoPricing(location string) (s dynamoPricing, err error) {
+	provisionedReadGroupDescription := "DynamoDB Provisioned Read Units"
+	provisionedWriteGroupDescription := "DynamoDB Provisioned Write Units"
+	onDemandReadGroupDescription := "DynamoDB PayPerRequest Read Request Units"
+	onDemandWriteGroupDescription := "DynamoDB PayPerRequest Write Request Units"
+
+	readCapacityUnit, err := getPrice(location, provisionedReadGroupDescription)
+
+	if err != nil {
+		return
 	}
+
+	writeCapacityUnit, err := getPrice(location, provisionedWriteGroupDescription)
+
+	if err != nil {
+		return
+	}
+
+	onDemandRead, err := getPrice(location, onDemandReadGroupDescription)
+
+	if err != nil {
+		return
+	}
+
+	onDemandWrite, err := getPrice(location, onDemandWriteGroupDescription)
+
```

Comment:
Go code is typically packed much more densely than other languages. These, and other spacing lines are normally omitted in Go. They're left in between sections of code which do different things, but omitted everywhere else.